### ISSUE TITLE
[BUG] Embed in Email tab freezes app for a few seconds

### ISSUE THREAD

**Issue by jobenjada** [formbricks/formbricks#1606]
### Issue Summary

Clicking on the embed in an email tab, the whole app freezes for a few seconds:

<img width="993" alt="image" src="https://github.com/formbricks/formbricks/assets/72809645/46086d26-0e34-4dd5-82b8-c3cc0d090e18">

### Steps to Reproduce

1. Create a survey
2. In the Share modal, select Embed in Email
3. See it freeze

### Expected behavior

Dont freeze

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [ ] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS:
- Node:
- npm:
```

**Comment by Sudipto-das** [comment]
i want to work on this bug

**Comment by Sudipto-das** [comment]
this happen bcz of heavy task performing on EmailTab component

**Comment by mattinannt** [comment]
I think a good option would be to add a loading indicator to show to the user that the content of the email tab is currently loading.

**Comment by Sudipto-das** [comment]
yes i already implement this but initially browser getting stuck and then show the loading indicator

**Comment by Inder782** [comment]
Hi it would be great if you can review my PR[https://github.com/formbricks/formbricks/pull/1622](url)
